### DISCUSSION TITLE
SystemVMs interfaces speed

### DISCUSSION THREAD

**Discussion by cdfgallo** [apache/cloudstack#9816] [Q&A] [answered]
Hi everyone, 

I was wondering if there is a way in Cloudstack to set the speed of the interfaces of systems to 10gbps in KVM environment.
Currently, even if hosts support 10gbps speed and VMs are using virtio drivers all interfaces are running at 1gbps.

Thank you in advance!

**Comment by weizhouapache** [comment]
@cdfgallo 
what's the speed of the linux bridges ?

```
# ethtool cloudbr0 |grep Speed
	Speed: 10000Mb/s
```

**Comment by cdfgallo** [comment]
@weizhouapache 
All bridges have 10gbps speed, here the example from our server

```
cloud08:~# ethtool cloudbr0 | grep Speed
        Speed: 10000Mb/s
```

**Reply by weizhouapache** [reply]
what's the MTU size ?
can you get 10Gbps on the host ? have you tested it ?

**Comment by cdfgallo** [comment]
Yes on the host I can have 10gbps, also if I force interfaces on a systemvm with command `ethtool -s eth0 speed 10000 duplex full` I can force the interface to 10gbps and it actually works, until the next reboot of course

**Reply by weizhouapache** [reply]
ok
is the command executed in the system vm ?
(eth0 in system vms is the link local interface)

**Reply by cdfgallo** [reply]
yes, of course it works for every interfaces. I tried installing iperf tool on a secstor VM and actually I can reach 10gbps after forcing speed via ethtool command

**Reply by weizhouapache** [reply]
that's great. 
it looks like all interfaces/bridges on the kvm hosts are configured correctly, except the interfaces inside the system vms.

do you get the same output inside the system vm ?
```
root@s-1147-VM:~# ethtool eth2
...
	Speed: Unknown!
	Duplex: Unknown! (255)
```

**Reply by cdfgallo** [reply]
yes, that's right. How can configure them ?

**Reply by rajujith** [reply]
Is it set to unlimited by default? 

![Screenshot 2024-10-17 at 1 05 07 PM](https://github.com/user-attachments/assets/8db71ca9-1899-4882-9272-e6f0980ef009)

**Reply by weizhouapache** [reply, marked as answer]
could it be caused by the linux bridge ? it works in half duplex mode

**Reply by cdfgallo** [reply]
found the issues that was in my configuration.
Now all works as expected
Thank you for your support :)

**Reply by weizhouapache** [reply]
> found the issues that was in my configuration. Now all works as expected Thank you for your support :)

good
can you share more details of your configuration ?
